### PR TITLE
3496 refactor api v1 imports

### LINKED ISSUE #3496: Refactor api/v1/imports

### PR DESCRIPTION
**Pull request by juanignaciosl**
@Kartones , @rafatower CR this (closes #3496). I've created `DataImportsService` to extract "external services" (resque) from the model and not adding logic to the controller, make inline comments about the design in order to discuss it :)

@javisantana could you CR the JS side, please?

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3542/))

File: app/services/carto/data_imports_service.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def stuck?(import)
+      # TODO: this kind of method is in the service because it requires communication with external systems (resque). Anyway, should some logic (state check, for example) be inside the model?
+      ![Carto::DataImport::STATE_ENQUEUED, Carto::DataImport::STATE_PENDING, Carto::DataImport::STATE_COMPLETE, Carto::DataImport::STATE_FAILURE].include?(import.state) &&
+      import.created_at < 5.minutes.ago &&
+      !running_import_ids.include?(self.id)
+    end
+
+    def running_imports_ids
+      Resque::Worker.all.map { |worker| worker.job["payload"]["args"].first["job_id"] rescue nil }.compact
```
Comment (rafatower) [inline app/services/carto/data_imports_service.rb]: interesting how you go to the queue to check running jobs. Are u sure this works all the time? do we want to have this kind of IPC?

File: spec/requests/api/json/imports_controller_shared_examples.rb [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    get api_v1_imports_index_url, params
+
+    response.code.should be == '200'
+
+    response_json = JSON.parse(response.body)
+    response_json.should_not be_nil
+    imports = response_json['imports']
+    imports.should have(1).items
+    Resque.inline = true
```
Comment (rafatower) [inline spec/requests/api/json/imports_controller_shared_examples.rb]: legacy code, maybe we can remove the inline = false/true

File: app/services/carto/data_imports_service.rb (hunk 1 above)
Comment (rafatower) [inline app/services/carto/data_imports_service.rb]: mmm I see it is copied from legacy code, so well tested :)

**Comment by rafatower** [comment]
:+1:

File: app/services/carto/data_imports_service.rb (hunk 1 above)
Comment (Kartones) [inline app/services/carto/data_imports_service.rb]: yes, is a :hankey: needed to check if process has died (due to Monit frecuently)

**Comment by javisantana** [comment]
javascript seems ok

File: spec/requests/api/json/imports_controller_shared_examples.rb (hunk 2 above)
Comment (Kartones) [inline spec/requests/api/json/imports_controller_shared_examples.rb]: mm careful, for testing all tests by default go as `inline=true`, this spec checks explicitly that the queue gets 1 item after posting to the endpoint.

I think is right here to leave it

**Comment by Kartones** [comment]
+1

File: spec/requests/api/json/imports_controller_shared_examples.rb (hunk 2 above)
Comment (juanignaciosl) [inline spec/requests/api/json/imports_controller_shared_examples.rb]: I'm afraid we can't (it fails). Anyway, I think inlining is great for testing this kind of logic.

**Comment by juanignaciosl** [comment]
@rafatower I just uploaded a improvement on what you suggested.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3543/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3582/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3604/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3607/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3608/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3611/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3621/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3623/))

**Comment by juanignaciosl** [comment]
@rafatower @Kartones my last commit, moving from `.find(id)` to `where(id: id).first` workarounded the [problem of nonexisting prepared statement](https://rollbar.com/vizzuality/CartoDB/items/6685/occurrences/5271121938/):

```
ActiveRecord::StatementInvalid: PG::Error: ERROR: prepared statement "a1" does not exist
: SELECT "data_imports".* FROM "data_imports" WHERE "data_imports"."id" = $1 LIMIT 1
... 23 non-project frames
24
File "/home/ubuntu/www/production.cartodb.com/releases/20150512111247/app/services/carto/data_imports_service.rb", line 31 in process_by_id
25
File "/home/ubuntu/www/production.cartodb.com/releases/20150512111247/app/controllers/carto/api/imports_controller.rb", line 13 in show
... 6 non-project frames
32
File "/home/ubuntu/www/production.cartodb.com/releases/20150512111247/app/controllers/application_controller.rb", line 54 in wrap_in_profiler
... 67 non-project frames
```

It looks like it needs more metadata fetching in order to guess pk column and something's broken there. I prefer moving on, since this looks quite dark.

Please CR this again before merging. You can test at ded03.

**Comment by rafatower** [comment]
:+1:

**Comment by Kartones** [comment]
:+1:

**Pull request merged**